module/VuFind/src/VuFind/Service/CurrencyFormatter.php

```diff
@@ -0,0 +1,106 @@
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Service;
+
+use NumberFormatter;
+
+/**
+ * Currency formatter
+ *
+ * @category VuFind
+ * @package  Service
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class CurrencyFormatter
+{
+    /**
+     * Default currency format (ISO 4217) to use.
+     *
+     * @var string
+     */
+    protected $defaultCurrency;
+
+    /**
+     * Number formatter.
+     *
+     * @var NumberFormatter
+     */
+    protected $formatter;
+
+    /**
+     * Constructor
+     *
+     * @param string $defaultCurrency Default currency format (ISO 4217) to use (null
+     * for default from locale)
+     */
+    public function __construct($defaultCurrency = null)
+    {
+        // Initialize number formatter:
+        $locale = setlocale(LC_MONETARY, 0);
+        $this->formatter = new NumberFormatter($locale, NumberFormatter::CURRENCY);
+
+        // Initialize default currency:
+        if (null === $defaultCurrency) {
+            $localeInfo = localeconv();
+            $defaultCurrency = isset($localeInfo['int_curr_symbol'])
+                ? trim($localeInfo['int_curr_symbol']) : '';
+        }
+        $this->defaultCurrency = empty($defaultCurrency) ? 'USD' : $defaultCurrency;
+    }
+
+    /**
+     * Convert currency from float to display format
+     *
+     * @param float  $number   The number to format
+     * @param string $currency Currency format (ISO 4217) to use (null for default)
+     *
+     * @return string
+     */
+    public function convertToDisplayFormat($number, $currency = null)
+    {
+        if (null === $currency) {
+            $currency = $this->defaultCurrency;
+        }
+        // Workaround for a problem in ICU library < 4.9 causing formatCurrency to
+        // fail if locale has comma as a decimal separator.
+        // (see https://bugs.php.net/bug.php?id=54538)
```

Comment:
bugs.php.net is not responding right now, so I can't look at the details of this... but I wonder if this is now so old that it is no longer an issue and this code can be simplified.

Reply:
Removed to code. Seems to be a fairly old ICU version that had it, and e.g. Ubuntu 18.04 LTS has a much newer one.

Reply:
Looks like the underlying ICU bug was fixed in 2018 in v49, so I agree that this should be safe to remove. I checked my PHP 7.3 installation on RHEL7 (which is about the oldest realistic platform that anyone would be trying to run VuFind 8.1 on), and it's using ICU v50.